apps/examples/testcase/ta_tc/audio/utc/utc_audio_main.c

```diff
@@ -638,43 +638,50 @@ static void utc_audio_pcm_readi_p(void)
 {
 	int fd;
 	char input_str[16];
-	ssize_t size;
-	ssize_t bytes;
-	int frames_read = 0;
-	unsigned int total_frames_read = 0;
-	unsigned int rate;
-	int remain;
 	int ret;
 	char *str;
 
-	size = pcm_get_buffer_size(g_pcm);
-	rate = pcm_get_rate(g_pcm);
-	bytes = rate * (pcm_format_to_bits(pcm_get_format(g_pcm)) / 8) * pcm_get_channels(g_pcm) * AUDIO_RECORD_DURATION;
+	char *buffer;
+	unsigned int size;
+	unsigned int frames_read;
+	unsigned int total_frames_read;
+	unsigned int bytes_per_frame;
+	int capturing = 1;
 
-	g_record_buffer = (char *)malloc(bytes);	//rate * bit * channels * duration
-	TC_ASSERT_NEQ("pcm_readi_p", g_record_buffer, NULL);
+	size = pcm_frames_to_bytes(g_pcm, pcm_get_buffer_size(g_pcm));
+	buffer = malloc(size);
+	TC_ASSERT_NEQ("pcm_readi", buffer, NULL);
+
+	fd = open(AUDIO_TEST_FILE, O_RDWR | O_CREAT);
+	TC_ASSERT_GT("pcm_readi", fd, 0);
```

Comment:
You should use TC_ASSERT_GT_CLEANUP in here.
Currently it returns without free allocated memory, "buffer" at failure case.
It causes a memory leak.

Reply:
You're right. I'll fix this.